pkg/api/types.go

```diff
@@ -1380,3 +1381,15 @@ const (
 
 	PortHeader = "port"
 )
+
+// ResourceStats represents compute resource usage metrics associated with a pod or a container.
+type ResourceStats struct {
+	// Detailed instantaneous metrics.
+	Detailed *cadvisor_api.ContainerInfo `json:"detailed,omitempty"`
```

Comment:
The intent is to not expose all the stats via the master since it will be a lot of data. The kubelet API will be extended soon to expose historical usage distribution, which is what @rjnagal is working on. That will be something we will end up exposing via the master API. The copying overhead should be minimal I hope.